CheckboxItem/CheckboxItem.js

```diff
@@ -81,14 +87,51 @@ const CheckboxItemBase = kind({
 		 */
 		icon: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
 
+		/**
+		 * Enables the "indeterminate" state.
+		 *
+		 * An indeterminate, mixed, or half-selected state is typically used in a hierarchy or group
+		 * to represent that some, not all, children are selected.
+		 *
+		 * NOTE: This does not prevent updating the `selected` state. Applications must control this
+		 * property directly.
+		 *
+		 * @type {Boolean}
+		 * @public
+		 */
+		indeterminate: PropTypes.bool,
+
+		/**
+		 * The icon to be used in the `indeterminate` state.
+		 *
+		 * May be specified as either:
+		 *
+		 * * A string that represents an icon from the [iconList]{@link agate/Icon.Icon.iconList},
+		 * * An HTML entity string, Unicode reference or hex value (in the form '0x...'),
+		 * * A URL specifying path to an icon image, or
+		 * * An object representing a resolution independent resource (See {@link ui/resolution})
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		indeterminateIcon: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+
 		/**
 		 * If true the checkbox will be selected.
 		 *
 		 * @type {Boolean}
 		 * @default false
 		 * @public
 		 */
-		selected: PropTypes.bool
+		selected: PropTypes.bool,
+
+		/**
+		 * Nodes to be inserted after the checkbox and before `children`.
+		 *
+		 * @type {Node}
+		 * @public
+		 */
+		slotBefore: PropTypes.node
 	},
 
 	defaultProps: {
```

Comment:
I don't think we need this anymore.

Reply:
I can remove `indeterminate` and `selected` since they are default to `false`, but we need `children` and `indeterminateIcon` since they are used in `computed` and `render` methods.

Reply:
I meant this `defaultProps` block is no needed anymore since we are removing `check` default value of `icon`.

Reply:
Oh... right. I will delete it